ports/nrf/supervisor/internal_flash.c

```diff
@@ -68,81 +67,14 @@ uint32_t supervisor_flash_get_block_size(void) {
 }
 
 uint32_t supervisor_flash_get_block_count(void) {
-    return ((uint32_t) __fatfs_flash_length) / FILESYSTEM_BLOCK_SIZE ;
-}
-
-#ifdef BLUETOOTH_SD
-STATIC void sd_flash_operation_start(void) {
-    sd_flash_operation_status = SD_FLASH_OPERATION_IN_PROGRESS;
-}
-
-STATIC sd_flash_operation_status_t sd_flash_operation_wait_until_done(void) {
-    while (sd_flash_operation_status == SD_FLASH_OPERATION_IN_PROGRESS) {
-        sd_app_evt_wait();
-    }
-    return sd_flash_operation_status;
-}
-#endif
-
-void supervisor_flash_flush(void) {
-    if (_flash_page_addr == NO_CACHE) return;
-
-    // Skip if data is the same
-    if (memcmp(_flash_cache, (void *)_flash_page_addr, FL_PAGE_SZ) != 0) {
```

Comment:
This old code checked for writing identical contents to the flash page, and skipped if that were true. I don't see the same check in your refactored code.

Reply:
Sorry, managed to break that while rebasing it I think ... have just pushed changes which puts it back together, and now I'm making clean and rerunning my tests!